controllers/workspace/provision/deployment.go

```diff
@@ -70,9 +70,24 @@ func SyncDeploymentToCluster(
 	podAdditions []v1alpha1.PodAdditions,
 	saName string,
 	clusterAPI ClusterAPI) DeploymentProvisioningStatus {
+
+	configmaps, err := getDevWorkspaceConfigmaps(workspace.Namespace, clusterAPI.Client)
+	if err != nil {
+		return DeploymentProvisioningStatus{
+			ProvisioningStatus: ProvisioningStatus{Err: err},
+		}
+	}
+
+	secrets, err := getDevWorkspaceSecrets(workspace.Namespace, clusterAPI.Client)
+	if err != nil {
+		return DeploymentProvisioningStatus{
+			ProvisioningStatus: ProvisioningStatus{Err: err},
+		}
+	}
+
 	// [design] we have to pass components and routing pod additions separately because we need mountsources from each
 	// component.
-	specDeployment, err := getSpecDeployment(workspace, podAdditions, saName, clusterAPI.Scheme)
+	specDeployment, err := getSpecDeployment(workspace, configmaps, secrets, podAdditions, saName, clusterAPI.Scheme)
```

Comment:
An alternate approach I would suggest here is to reuse `volumes` and volumeMounts` in `podAdditionsList`. This function (`getSpecDeployment`) will already automatically add all volumes/volumeMounts to the workspace deployment/containers, so we're kind of duplicating work right now.

In other words, we could have `getDevWorkspaceSecrets` and `getDevWorkspaceConfigMaps` return `*v1alpha1.podAdditions`, and then just append those to the `podAdditions` we have here. Most of the code required is already present, so this would mainly be reorganizing things.

Reply:
Got it! Finally. Much cleaner! Thanks @amisevsk